src/values/dynamic-item-value.js

```diff
@@ -0,0 +1,39 @@
+import ChildElements from '../elements/child-elements';
+import FormValues from './form-values';
+
+export default class DynamicItemValue {
+  constructor(dynamicValue, attributes) {
+    this.dynamicValue = dynamicValue;
+    this._metadataJSON = attributes.metadata;
+    this._elementsJSON = attributes.elements;
+    this._valuesJSON = attributes.values;
+  }
+
+  toJSON() {
+    const json = {};
+
+    json.metadata = this._metadataJSON || null;
+    json.elements = this._elementsJSON || null;
+    json.values = this.values.toJSON() || null;
+
+    return json;
+  }
+
+  get id() {
+    return this.metadata.id;
+  }
+
+  get metadata() {
+    return this._metadataJSON;
+  }
+
+  get values() {
```

Comment:
I'm not certain yet whether we'll want to also provide a `formValues` getter that does this same thing. But that would make this fill the same interface as a Feature/Record

Reply:
punting for now